src/test/java/io/jenkins/plugins/analysis/warnings/IssuesRecorderITest.java

```diff
@@ -71,19 +86,370 @@ public void shouldCreateUnstableResult() {
+        tableCells = row.getCells();
+
+        assertThat(tableCells.get(2).getTextContent()).isEqualTo("Normal");
+
+        simian.setNormalThreshold(5);
+
+        table = getIssuesTable(project, "simian");
+        tableRows = table.getBodies().get(0).getRows();
+        row = tableRows.get(0);
+        tableCells = row.getCells();
+
+        assertThat(tableCells.get(2).getTextContent()).isEqualTo("Low");
+    }
+
+    /**
+     * Verifies that the priority links are redirecting to a filtered side,
+     * showing only the warnings of this priority.
+     */
+    @Test
+    public void priorityLinksShouldOpenFilteredSite() {
+        FreeStyleProject project = createJobWithWorkspaceFile("duplicateCode/cpd.xml");
+        Cpd cpd = new Cpd();
+        cpd.setNormalThreshold(2);
+        cpd.setHighThreshold(4);
+        enableWarnings(project, cpd);
+
+        List<HtmlTableRow> tableRows = getIssueTableRows(project);
+
+        HtmlTableRow rowWithLowPriorityWarning = tableRows.get(5);
+        HtmlTableRow rowWithNormalPriorityWarning = tableRows.get(0);
+        HtmlTableRow rowWithHighPriorityWarning = tableRows.get(2);
+
+        HtmlPage lowPriorityPage = clickOnLink(rowWithLowPriorityWarning.getCell(2).getFirstElementChild());
+        HtmlPage normalPriorityPage = clickOnLink(rowWithNormalPriorityWarning.getCell(2).getFirstElementChild());
+        HtmlPage highPriorityPage = clickOnLink(rowWithHighPriorityWarning.getCell(2).getFirstElementChild());
+
+
+        HtmlHeading5 heading = (HtmlHeading5) lowPriorityPage.getElementById("number-priorities");
+        checkAmountOfPriorityWarnings(heading,6,0,0);
+
+        heading = (HtmlHeading5) normalPriorityPage.getElementById("number-priorities");
+        checkAmountOfPriorityWarnings(heading,0,9,0);
+
+        heading = (HtmlHeading5) highPriorityPage.getElementById("number-priorities");
+        checkAmountOfPriorityWarnings(heading,0,0,5);
+    }
+
+    /**
+     * Verifies that the source code links are redirecting to a side displaying the source code.
+     */
+    @Test
+    public void sourceCodeLinksShouldWork(){
+        FreeStyleProject project = createJobWithWorkspaceFile("duplicateCode/cpd.xml");
+        Cpd cpd = new Cpd();
+        cpd.setNormalThreshold(2);
+        cpd.setHighThreshold(4);
+        enableWarnings(project, cpd);
+
+        List<HtmlTableRow> tableRows = getIssueTableRows(project);
+
+        HtmlPage sourceCodePage = clickOnLink(tableRows.get(0).getCell(1).getFirstElementChild());
+
+        DomElement tableElement = sourceCodePage.getElementById("main-panel");
+
+        String htmlFile = null;
+        try {
+            htmlFile = Streams.asString(getTestResourceClass()
+                    .getResourceAsStream("duplicateCode/expected_html_code_block.html")).trim();
+        }
+        catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        assertThat(tableElement.asText()).isEqualTo(htmlFile);
```

Comment:
Die Assertion schlägt bei mir fehl. Wenn ich den Test richtig verstehe, navigiert er über den Source Code Link in die View mit dem Source Code. Aber in Ihrem PR ist ja gar kein Source Code File mit drin, das in den Workspace kopiert wird. Ohne das wird ja auch nie ein File angezeigt. Fehlt da noch was im PR?

Reply:
Ich schau es mir gleich nochmal an, danke für den hinweis.

Reply:
Aus IT-Sicherheitssicht müsste da eine Validierung eingebaut werden, ob sich der Pfad im workspace befindet, sonst führt das zu einer Sicherheitslücke.
Ich hab mal schnell nen test gemacht, bei dem ich die Pfade in der xml auf /etc/passwd geändert hab, was mir die passwd Datei angezeigt hat.
Bei der Path-Validierung muss auch beachtet werden, ob das source-file z.B. ein symlink ist, der aus dem workspace heraus führt, falls der Code diesen folgen würde.